chronosphere/tfschema/trace_common.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2024 Chronosphere Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfschema
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/enum"
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/prettyenum"
+)
+
+var TraceSearchFilterSchema = &schema.Schema{
+	Type:     schema.TypeList,
+	MinItems: 1,
+	MaxItems: 1,
+	Required: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"trace": TraceFilterSchema,
+			"span":  TraceSpanFilterListSchema,
+		},
+	},
+}
+
+var TraceFilterSchema = &schema.Schema{
+	Type:     schema.TypeList,
+	MinItems: 0,
+	MaxItems: 1,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"duration": TraceDurationFilterSchema,
+			"error":    TraceBoolFilterSchema,
+		},
+	},
+}
+
+var TraceSpanFilterListSchema = &schema.Schema{
+	Type:     schema.TypeList,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"match_type": {
+				Type:             schema.TypeString,
```

Comment:
I'll look into follow-on; I carried over this logic from one of the two consolidated types because of some clear-looking intent in some of the internal tests related to it (e.g., lowercasing of enum values, etc)

Reply:
will double check though in the follow-on PR for the above comment. `prettyenum` seemed "newer" to me but it's possible I misunderstood its original purpose